etrago/tools/execute.py

```diff
@@ -400,11 +455,22 @@ def drop_foreign_components(network):
             drop_foreign_components(network)
 
     # Set slack bus
+    ac_bus = network.buses[network.buses.carrier == "AC"]
+    network.generators.control[network.generators.bus.isin(ac_bus.index)] = "PV"
+    network.generators.control[network.generators.carrier == "load shedding"] = "PQ"
     network = set_slack(network)
+    foreign_buses = network.buses[network.buses.country != "DE"].index
+    network.generators.control[(network.generators.bus.isin(foreign_buses))
+                               &(network.generators.bus.isin(ac_bus.index))] = "Slack"
 
     # execute non-linear pf
     pf_solution = network.pf(network.snapshots, use_seed=True)
 
+    # Keep results of the network inside Germany
+    sub_net = str(network.buses.sub_network.value_counts().argmax())
+    for i in pf_solution.keys():
+        pf_solution[i] = pf_solution[i][sub_net]
+
```

Comment:
I'm not sure if I completely understand these lines. Does it mean that the pf is also executed for smaller subnetworks that only consist of foreign buses? 
Before, we dropped the other subnetworks to avoid feasibility problems with the pf (see lines 451 - 455).

Reply:
> I'm not sure if I completely understand these lines. Does it mean that the pf is also executed for smaller subnetworks that only consist of foreign buses? Before, we dropped the other subnetworks to avoid feasibility problems with the pf (see lines 451 - 455).

Now the PF is only applied to the main network.